### PR TITLE
:recycle: (tooltip) refactored tooltip to react-aria

### PR DESCRIPTION
**Pull request by MatissJanis**
Related to: https://github.com/actualbudget/actual/issues/2091

This refactors the `Tooltip` component we use internally to instead use react-aria. Initially I am refactoring only the `Notes` button, but the goal is to continue this migration after the initial PR is released.

### CODE REVIEW

File: packages/desktop-client/src/components/NotesButton.tsx [hunk 1]
```diff
@@ -1,116 +1,87 @@
...
+            ...(hasNotes && { display: 'flex !important' }),
+            ...(isOpen && { color: theme.buttonNormalText }),
+          }}
+          onClick={() => setIsOpen(true)}
+        >
+          <SvgCustomNotesPaper style={{ width, height }} />
+        </Button>
+      </View>
+
+      <Popover
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/NotesButton.tsx:76]: So the `NotesButton` component now uses both Tooltip + Popover.

Tooltip - should be used for "hover" effects. They appear on hover. And disappear when moving mouse cursor away.

Popover - should be used for more permanent overlays that people can interact with (i.e. the textbox for editing notes).

Previously we were mixing these two concepts together, but I think it makes sense to slightly refactor this to separate the concerns in 2x different components.

**Comment by youngcw** [comment]
There used to be a slight delay in showing the note hover tooltip.  I think that was nice so just moving the cursor past the note icon wouldn't do anything but a deliberate hover would.

**Comment by MatissJanis** [comment]
> There used to be a slight delay in showing the note hover tooltip. I think that was nice so just moving the cursor past the note icon wouldn't do anything but a deliberate hover would.

Good shout! Patched this now.

**Comment by youngcw** [comment]
Now im getting that issue where the tooltips don't go away if you mouse over quickly.  The delay to open is good though.

**Comment by MatissJanis** [comment]
Eagle eye! Refactored a bit to patch this issue :)

**Comment by youngcw** [comment]
Nice. All seems good now

File: yarn.lock [hunk 2]
```diff
@@ -3157,6 +3167,16 @@ __metadata:
   languageName: node
   linkType: hard
 
+"@internationalized/message@npm:^3.1.2":
```
Comment (twk3) [inline yarn.lock:3170]: @MatissJanis can you run a `yarn dedupe` so we aren't collecting multiple patch releases of the same libs in the node modules. (will also clean up the past few months, I last had a dedupe committed in jan I think)

Comment (MatissJanis) [reply yarn.lock:2201]: Good idea! https://github.com/actualbudget/actual/pull/2493/commits/6fcde538dc4e42ce46ac3a35b8420ce383959607

**Comment by joel-jeremy** [comment]
This is nice! Would it be also a good idea to migrate some of our existing components to react aria components?

**Comment by joel-jeremy** [comment]
Clicking on the group notes button expands/collapses the group

File: packages/desktop-client/src/components/NotesButton.tsx [hunk 3]
```diff
@@ -1,116 +1,87 @@
...
+  tooltipPosition = 'bottom start',
   style,
 }: NotesButtonProps) {
-  const [hover, setHover] = useState(false);
-  const tooltip = useTooltip();
-  const data = useLiveQuery(() => q('notes').filter({ id }).select('*'), [id]);
-  const note = data && data.length > 0 ? data[0].note : null;
+  const triggerRef = useRef(null);
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const data = useLiveQuery<NoteEntity[]>(
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/NotesButton.tsx:35]: This would be nice to have as a hook, but that can be done in a separate PR

**Comment by MatissJanis** [comment]
> Clicking on the group notes button expands/collapses the group

Good catch. This is now patched.

File: packages/desktop-client/src/components/NotesButton.tsx (hunk 3 above)
Comment (MatissJanis) [reply packages/desktop-client/src/components/NotesButton.tsx:35]: Definitely agreed. But lets try to keep the scope of changes small (i.e. this PR).

**Review by twk3 — APPROVED** [review]
Looks good. :+1:

**Pull request merged**